src/biogeophys/SnowCoverFractionMod.F90

```diff
@@ -0,0 +1,785 @@
+module SnowCoverFractionMod
```

Comment:
It feels like most of this module is infrastructure.  Is there any way to make the 'science' code more front-and-center?

Reply:
I have a few ideas, and would be interested to hear what you think of them, or if you have other thoughts:

1. I could move the definition of the abstract snow_cover_fraction_type and the whole abstract interface block near the top of this module into its own module. This is what I have done for other object-oriented parts of the code; the downside is just having more files. That would remove about 85 lines from the start of the file.

2. I could also move the factory method (CreateAndInitScfMethod) into its own module. Again, this is what I have done for other object-oriented parts of the code; the downside is again having more files. (Unfortunately, this cannot be combined with the module for the abstract type, because that introduces a circular dependency.) This would remove about 50 lines of code. Alternatively, I could move this method to the bottom of the file.

3. I could move the various science routines to above the infrastructurey routines that are just used in initialization. If I do this, I'd like to know if you'd prefer:

(a) The different parameterizations each get their own file

(b) The different parameterizations stay together in a single file, with all subroutines for a given parameterization grouped together (as they are now), but just change the order of these routines within each grouping so that the science routines come first.

(c) The different parameterizations stay together in a single file, with two sets of groupings: first we have all of the science routines for NY07 followed by all of the science routines for SL12; then we have all of the initialization routines for NY07 followed by all of the initialization routines for SL12.

(d) The different parameterizations stay together in a single file, with the groupings flipped around: first we have UpdateSnowDepthAndFracNY07 followed by UpdateSnowDepthAndFracSL12; then we have AddNewsnowToIntsnowNY07 followed by AddNewsnowToIntsnowSL12; etc.

I am totally happy to do any of these, so please let me know what you'd find most helpful.

Reply:
I think personally I like the idea of an abstract interface for the infrastructure, and then a ny07 implementation and clm5 (still not sure I like calling it the clm5 version) implementation. I think it's OK to have the different implementations in different files, and maybe better actually. The science is then pretty much all front and center in either of the implementations. I picture it can be useful to compare the different implementations, but then having them in different files is actually helpful. Since, the implementations are quite different it seems unlikely that you'd have shared code. And if you do you could probably put in in a base class.

Reply:
Thanks for your thoughts @ekluzek . You raise good points about the advantages of having separate files. So if I understand correctly, it sounds like your preferences on (3) would be (a) (which I think would also imply doing (1) & (2)) then (b). I'll talk more with @swensosc about this next week to see if he agrees.